### ISSUE TITLE
Service port and deployment port do not match

### ISSUE THREAD

**Issue by alexanderbuhler** [kiali/kiali#1228]
**Describe the bug**

Probably not really a bug, but something is obviously wrong with my configuration.
Kiali says the port configuration of my service is not valid.

<img width="406" alt="kiali_invalid_port_config" src="https://user-images.githubusercontent.com/5637707/60877617-77050300-a23e-11e9-97ad-6684fb32ede6.png">

I mean obviously, the service and container port don't match, but that's what a service is for at the end? Why is it a problem from kialis point of view?

So the service is configured to listen on port 80 TCP. Port ist named http. Mapped to targetPort 8080.
```yaml
spec:
  selector:
    app: auth
  ports:
    - protocol: TCP
      port: 80
      targetPort: 8080
      name: http
```
The corresponding pod listens on 8080.

```yaml
        ports:
        - containerPort: 8080
```

**Versions used**
Kiali: 0.20.0
Istio: 1.2.2
Kubernetes flavour and version: 1.13.7

**To Reproduce**

1. Visit "Services" in the UI
2. View details of a service

**Expected behavior**
Port configuration is recoginzed as valid.

**Comment by jotak** [comment]
@alexanderbuhler this sounds indeed like a curious validation performed by Kiali, however it relates to a real problem, see https://issues.jboss.org/browse/KIALI-2454 and https://github.com/istio/istio/issues/9504

If at some point this isn't a problem anymore in Istio (or if somehow we are misinterpreting the problem) then we should reconsider this validation.

**Comment by lucasponce** [comment]
Please @burmanm, can you take a look to this one ?

**Comment by alexanderbuhler** [comment]
@jotak well everything seems to be working fine besides the complaining by kiali... gateways/vservices/services/pods

**Comment by burmanm** [comment]
https://issues.jboss.org/browse/KIALI-3115

**Comment by burmanm** [comment]
.. although, with the ClusterIP=None that above example is actually invalid and should be flagged as invalid service (per Kubernetes types documentation - but real world might be different)

**Comment by huegelc** [comment]
Should this bug be fixed? I still experience this behaviour 

```
Kiali-ui:  0.20.0 (87ca7062921e2aa857c6322e9ed3f0882f089922)
Kiali: v0.20.0 (140969bc2ef1a77ca7387b47f8b23dcbd4818bd0)

Components

Istio:    1.2.4 
Prometheus: 2.8.0 
Kubernetes: v1.14.1 
```

**Comment by jshaughn** [comment]
@huegelc I believe the fix is in the Kiali 1.2 release.
